packages/fxa-settings/src/pages/Signup/ConfirmSignupCode/index.tsx

```diff
@@ -91,63 +82,56 @@ const ConfirmSignupCode = (_: RouteComponentProps) => {
     'Confirmation code is required'
   );
 
-  // When the user types in the code input field, all banners and tooltips should be cleared
-  // Timeout is added to reduce jankiness, but does not include a smooth hiding effect.
-  useEffect(() => {
-    if (clearMessages) {
-      const timer = setTimeout(() => {
-        setCodeErrorMessage('');
-        setBanner({ type: undefined, children: undefined });
-        setClearMessages(false);
-      }, CLEAR_MESSAGES_TIMEOUT);
-      return () => clearTimeout(timer);
-    }
-    return;
-  }, [clearMessages]);
-
-  // Hide the ResendCode button after too many attempts. Redisplay button after a delay.
-  useEffect(() => {
-    if (resendCodeCount > 3) {
-      const timer = setTimeout(() => {
-        setResendCodeCount(0);
-      }, RESEND_CODE_TIMEOUT);
-      return () => clearTimeout(timer);
-    }
-    return;
-  }, [resendCodeCount]);
-
   async function handleResendCode() {
     try {
       await account.sendVerificationCode();
-      setResendCodeCount(resendCodeCount + 1);
-      setResendStatus(ResendStatus['sent']);
-    } catch (e) {
-      setResendStatus(ResendStatus['error']);
+      // if resending a code is succesful, clear any banner already present on screen
+      if (resendStatus !== ResendStatus['sent']) {
+        setBanner({
+          type: undefined,
+          children: undefined,
+        });
+        setResendStatus(ResendStatus['sent']);
+      }
+    } catch (error) {
+      setResendStatus(ResendStatus['not sent']);
+      const localizedErrorMessage = getLocalizedErrorMessage(
+        ftlMsgResolver,
+        error
+      );
+      setBanner({
+        type: BannerType.error,
+        children: <p>{localizedErrorMessage}</p>,
+      });
     }
   }
 
-  function alertSuccessAndGoForward() {
-    alertBar.success(
-      ftlMsgResolver.getMsg(
-        'confirm-signup-code-success-alert',
-        'Account confirmed successfully'
-      )
-    );
-    // TODO in FXA-6519 redirect elsewhere if relying party
-    navigate('/settings', { replace: true });
-  }
-
   async function verifySession(code: string) {
     logViewEvent(`flow.${viewName}`, 'submit', REACT_ENTRYPOINT);
     try {
-      const newsletterSlugs = location.state?.selectedNewsletterSlugs;
-      const hasSelectedNewsletters =
-        newsletterSlugs && newsletterSlugs.length > 0;
+      const hasSelectedNewsletters = newsletters && newsletters.length > 0;
+
+      const options = {
```

Comment:
@LZoog does this updated version feel clearer to you?

Reply:
LGTM!